internal/atls/atls.go

```diff
@@ -347,7 +348,7 @@ func (c *clientConnection) verify(rawCerts [][]byte, verifiedChains [][]*x509.Ce
 func (c *clientConnection) getCertificate(cri *tls.CertificateRequestInfo) (*tls.Certificate, error) {
 	priv := c.privKey
 
-	if priv == nil {
+	if priv == nil || reflect.ValueOf(priv).IsNil() {
```

Comment:
```suggestion
	if priv == nil || priv != (*ecdsa.PrivateKey)(nil) {
```
we don't want reflect, I think we could compare withe the typed nil pointer here instead?

Reply:
Yes, but that wouldn't cover other key types of course. But I think I'd prefer that too.